src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,935 @@
+                    getJsonProperty(jsonNode, "versionInfo").asText());
+            statistics.put("Applications#" + getJsonProperty(jsonNode, "appName").asText() + " Last Updated",
+                    new Date(getJsonProperty(jsonNode, "lastUpdatedOn").asLong()).toString());
+        });
+    }
+
+    /**
+     * Retrieve list of active sessions on the device
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    public void getSessions(Map<String, String> statistics) throws Exception {
+        JsonNode response = doGet(buildHttpUrl(SESSIONS), JsonNode.class);
+        if (response.get("sessionList") == null) {
+            return;
+        }
+        response.get("sessionList").forEach(jsonNode -> {
+            statistics.put(String.format("Active Sessions#%s : %s : %s(%s)", getJsonProperty(jsonNode, "userId").asText(),
+                    getJsonProperty(jsonNode, "role").asText(), getJsonProperty(jsonNode, "location").asText(),
+                    getJsonProperty(jsonNode, "clientType").asText()),
+                    String.format("%s, %s", (getJsonProperty(jsonNode, "isConnected").asBoolean() ? "CONNECTED" : "NOT CONNECTED"),
+                            (getJsonProperty(jsonNode, "isAuthenticated").asBoolean() ? "AUTHENTICATED" : "NOT AUTHENTICATED")));
+        });
+    }
+
+    /**
+     * Retrieve list of device's microphones and it's information/state
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    public void getMicrophones(Map<String, String> statistics) throws Exception {
+        ArrayNode response = doGet(buildHttpUrl(MICROPHONES), ArrayNode.class);
+        response.forEach(jsonNode -> {
+            statistics.put(String.format("Microphones#Microphone %s Name", getJsonProperty(jsonNode, "number").asText()),
+                    getJsonProperty(jsonNode, "typeInString").asText());
+            statistics.put(String.format("Microphones#Microphone %s State", getJsonProperty(jsonNode, "number").asText()),
+                    getJsonProperty(jsonNode, "state").asText());
+            statistics.put(String.format("Microphones#Microphone %s Type", getJsonProperty(jsonNode, "number").asText()),
+                    getJsonProperty(jsonNode, "type").asText());
+            statistics.put(String.format("Microphones#Microphone %s HW Version", getJsonProperty(jsonNode, "number").asText()),
+                    getJsonProperty(jsonNode, "hwVersion").asText());
+            statistics.put(String.format("Microphones#Microphone %s SW Version", getJsonProperty(jsonNode, "number").asText()),
+                    getJsonProperty(jsonNode, "swVersion").asText());
+            statistics.put(String.format("Microphones#Microphone %s Mute", getJsonProperty(jsonNode, "number").asText()),
+                    String.valueOf("0".equals(getJsonProperty(jsonNode, "mute").asText())));

```

Comment:
isn't "getJsonProperty(jsonNode, "number")" (which can be extracted into local variable) putting same value for "Name", "State",  "Type", "HW Version", "SW Version", and "Mute"? maybe I'm missing something...lets talk about this too, so I understand how getJsonProperty() works